doc: The commit adds Azure Streams PrivateLink documentation and examples (#3299)
mongodb/terraform-provider-mongodbatlas: docs/resources/stream_privatelink_endpoint.md

### Required

> - `project_id` (String) Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.<br>**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group or project id remains the same. The resource and corresponding endpoints use the term groups.

Reviewer: Interesting usage of `<br>` inside attribute description!

- `provider_name` (String) Provider where the Kafka cluster is deployed. Valid values are AWS and AZURE.

> - `vendor` (String) Vendor that manages the Kafka cluster. The following are the vendor values per provider:<br>- MSK and CONFLUENT for the AWS provider.<br>- EVENTHUB and CONFLUENT for the AZURE provider.<br>**NOTE** Omitting the vendor field will default to using the EVENTHUB vendor for the AZURE provider.

Reviewer 2: ```suggestion
- `vendor` (String) Vendor that manages the Kafka cluster. The following are the vendor values per provider:<br>- MSK and CONFLUENT for the AWS provider.<br>- EVENTHUB and CONFLUENT for the AZURE provider.<br>**NOTE** If omitted, defaults to the EVENTHUB vendor for the AZURE provider.
```

### Optional

- `arn` (String) Amazon Resource Name (ARN). Required for AWS Provider and MSK vendor.
- `dns_domain` (String) The domain hostname. Required for the following provider and vendor combinations:<br>- AWS provider with CONFLUENT vendor.<br>- AZURE provider with EVENTHUB or CONFLUENT vendor.

> - `dns_sub_domain` (List of String) Sub-Domain name of Confluent cluster. These are typically your availability zones. Required for AWS Provider and CONFLUENT vendor, if your AWS CONFLUENT cluster doesn't use subdomains, you must set this to the empty array [].

Reviewer 2: ```suggestion
- `dns_sub_domain` (List of String) Sub-Domain name of Confluent cluster. These are typically your availability zones. Required for AWS Provider and CONFLUENT vendor. If your AWS CONFLUENT cluster doesn't use subdomains, you must set this to the empty array [].
```
